test/span.js

```diff
@@ -182,6 +186,121 @@ describe('span should', () => {
         assert.isOk(unnormalizedKey in Span._getBaggageHeaderCache());
     });
 
+    describe('adaptive sampling tests for span', () => {
+        let options = [
+            { desc: 'sampled: ', sampling: true, reportedSpans: 1 },
+            { desc: 'unsampled: ', sampling: false, reportedSpans: 0}
+        ];
+        _.each(options, (o) => {
+            it (o.desc + 'should save tags, and logs on an unsampled span incase it later becomes sampled', () => {
+                let reporter = new InMemoryReporter();
+                let tracer = new Tracer(
+                    'test-service-name',
+                    reporter,
+                    new ConstSampler(false),
+                    { logger: new MockLogger() }
+                );
+                let span = tracer.startSpan('initially-unsampled-span');
+                span.setTag('tagKeyOne', 'tagValueOne');
+                span.addTags({
+                    'tagKeyTwo': 'tagValueTwo'
+                });
+                span.log({'logkeyOne': 'logValueOne'});
+
+                tracer._sampler = new ConstSampler(o.sampling);
+                span.setOperationName('sampled-span');
+                span.finish();
+
+                assert.isOk(deepEqual(span._tags[0], {key: 'tagKeyOne', value: 'tagValueOne'}));
+                assert.isOk(deepEqual(span._tags[1], {key: 'tagKeyTwo', value: 'tagValueTwo'}));
+                assert.isOk(deepEqual(span._logs[0].fields[0], {key: 'logkeyOne', value: 'logValueOne'}));
+                assert.equal(reporter.spans.length, o.reportedSpans);
+            });
+        });
+
+        describe('Finalize spans that', () => {
+            it ('inherit sampling decision is finalized', () => {
```

Comment:
> To me the purpose of "describe" and "it" isn't to read like a literal sentence, 

the frameworks built to support BDT have certain conventions about how the tests should be written. This describe/it convention is nearly uniform across many languages. Your argument is essentially to abuse the framework by describing tests in a completely different way, which as a result reads completely weird in the source code. Why use a BDT notation in that case?

> "describe span used as a parent it should be finalized"

yes, because it's not meant to be read as a single line, but as a group and items:
  * describe "span used as a parent"
     * it "should be finalized"
     * it "should do something else"